src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -703,12 +698,55 @@ private static FormulaValue Substitute(IRContext irContext, FormulaValue[] args)
                 instanceNum = (int)nv.Value;
             }
 
+            // Compute max possible memory this operation may need.
```

Comment:
@bryancroteau-MSFT  - We should apply these sort of pre-checks to our other string,table functions.